test/modules/obsagent-client/src/config.rs

```diff
@@ -0,0 +1,173 @@
+        } else if push_period == 0.0 {
+            return Err(ConfigError::InvalidZeroArgVal(push_period));
+        }
+
+        Ok(OTelConfig {
+            push_period,
+            otlp_endpoint,
+        })
+    }
+}
+
+pub fn init_config() -> Result<Config, ConfigError> {
+    let matches = App::new("obs_agent_client")
+        .arg(
+            Arg::with_name("update-period")
+                .short("u")       
+                .long("update-period")
+                .takes_value(true)
+                .help("Period in seconds betweeen successive updates of each instrument with a new metric measurement.")
+        )
+        .arg(
+            Arg::with_name("push-period")
+                .short("p")
+                .long("push-period")
+                .takes_value(true)
+                .help("Period in seconds between successive pushes of  measurements  out of the client.")
+        )
+        .arg(
+            Arg::with_name("otlp-endpoint")
+                .short("e")
+                .long("otlp-endpoint")
+                .takes_value(true)
+                .help("Endpoint to which OTLP messages will be sent.")
+        )
+        .get_matches();
+
+    let push_period = std::env::var("PUSH_PERIOD")
+        .map_or_else(
+            |_e| Ok(value_t!(matches.value_of("push-period"), f64).unwrap_or(5.0)),
+            |v| v.parse(),
+        )
+        .map_err(ConfigError::FloatArgParseError)?;
+    let otlp_endpoint = std::env::var("OTLP_ENDPOINT")
+        .map_or_else(
+            |_e| {
+                Ok(matches
+                    .value_of("otlp-endpoint")
+                    .unwrap_or("http://localhost:4317")
+                    .to_string())
+            },
+            |v| v.parse(),
+        )
+        .map_err(ConfigError::StringArgParseError)?;
+    let otel_config = OTelConfig::new(push_period, otlp_endpoint)?;
+
+    let prom_config = PromConfig {
+        endpoint: std::env::var("PROMETHEUS_ENDPOINT")
+            .map_or_else(
+                |_e| {
+                    Ok(matches
+                        .value_of("prometheus-endpoint")
+                        .unwrap_or("127.0.0.1:9600")
+                        .to_string())
+                },
+                |v| v.parse(),
+            )
+            .map_err(ConfigError::StringArgParseError)?,
+    };
+
+    let update_period = std::env::var("UPDATE_PERIOD")
+        .map_or_else(
+            |_e| Ok(value_t!(matches.value_of("update-period"), f64).unwrap_or(1.0)),
+            |v| v.parse(),
+        )
+        .map_err(ConfigError::FloatArgParseError)?;
+    let config = Config::new(update_period, prom_config, otel_config)?;
+    Ok(config)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_invalid_zero_push_period() {
+        let otel_config = OTelConfig::new(0.0, String::from("localhost:4317"));
+        assert!(otel_config.is_err());
+    }
+
+    #[test]
+    fn test_invalid_negative_push_period() {
```

Comment:
1. Because we have already thought the scenario of f64:epsilon value input, we might as well commemorate it here in the test as well. (In case someone in the future change from period to be "rate" or add some computation down the road that will encounter underflow).
2. Likewise for `f64::INFINITY`

Reply:
Added f64::EPSILON and f64::INFINITY test cases to config.rs.